third_party/CMakeLists.txt

```diff
@@ -363,3 +363,8 @@ add_library(opencl INTERFACE)
 set(OPENCL_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/opencl)
 target_sources(opencl INTERFACE ${OPENCL_INCLUDE_DIR}/CL/opencl.h)
 target_include_directories(opencl INTERFACE ${OPENCL_INCLUDE_DIR})
+
+# Catch2
+add_subdirectory(catch2)
+set_property(TARGET Catch2 PROPERTY FOLDER "third_party")
```

Comment:
The other 3rd party modules are in the "ThirdParty" folder, would be good to have this consistent.

![image](https://user-images.githubusercontent.com/10283127/236866583-b4ef68c4-2c40-404f-912f-292e4ea5ff58.png)

Reply:
I would prefer keeping with the idea that header files are in the same folder as source files.  The reasoning behind this is the tooling assumes it and it takes less work to keep things working.  For example in CMake, you only need to include the source filename in the target description and it will find the header file automatically if it's in the same directory for purposes of determining if the file has changed etc.

Reply:
I'm not sure I quite understand where you are coming from. In V1.0 we still had to manually add headers to source groups for them to show up in IDEs. In the v2.0 branch I do not manually add any headers and recompilation has been fine with CMake on all platforms. I often recompile after small changes so I would like to think that I have tested this case thoroughly - any steps I can reproduce to hit this recompilation mishap?

The biggest issues I've had are with solution compatibility with Visual Studio. I haven't tried this on other IDEs. If this looks a bit off in your IDE of choice could you share a screenshot?

In terms of the header folder structure, it may seem a little overkill right now. However, when this folder structure grows the change is a little more natural. I am happy to drop the extra `core` / `components` subfolders. But these have helped reduce header collisions in my platform migration branch (that's a work in progress). It is a similar pattern to `#inlcude <spdlog/fmt/fmt.hpp>` the extra folders help "namespace" the headers.

Very stylistic choices, so let me know more about what sort of styles you think may work better and ill be happy to try and incorporate them.

Reply:
Adding headers to sourcegroups in order for an IDE to work is a bug in the IDE.  The problem with the work around of including the header files in the source groups is that it means that cmake will erroniously treat those header files as if they were source files and try to compile them.  Most of the time; this isn't a problem because not a lot of code goes into header files which, when compiled, creates bugs.  However, it is incorrect and it causes slow down in compile time and configuration time.  To confirm this, take a look in the .git folder for your listing of dependencies for the project.
When CMake goes through the creation of the dependencies, you want it to realize that header files are dependencies and not source files.  This way, if you touch a header file without listing it; it will cause a recompile correctly.

Visual Studio has recently received support for loading cmake projects directly.  I know that VS has recently been trying to fix some of these IDE bugs.  You should find that both XCode and CLion or (Android Studio) should behave correctly with regards to header files.  I don't know if it would help to try loading the project in VS as a cmake project rather than using kitware's VS generator command step.  However, this should hopefully be the correct thing to do with CMake projects.
If we don't get a good VS style solution; we can continue the work-around; but someday in the future we should review the decision again.  The likelihood of there being a real compilation bug that is caused by this is remote.

Something that might be a happy middle ground for the folder structure is to use header only target libraries.  Each folder has it's own CMakeLists.txt and the include directory merely sets the headers.  Then to use them; make them dependencies of the consuming target in CMake.  You can then just say "#include <fmt.hpp>"  or similar without worry about the folder structure.

Reply:
With ``vkb__register_component` we can tweak this to use both. Interface for headers and static or shared for headers + source. I do want to try and keep the public and private headers as separate as possible as it encourages only exposing the public stuff in the future. One issue we can see in samples and many other projects is how coupled these different areas get over time. So separation by default can be a good thing

Reply:
Any actions from this conversation? Other than Saschas `third_party` rename